apps/whale-api/src/module.api/governance.controller.e2e.ts

```diff
@@ -0,0 +1,382 @@
+      next: resultPage1.page?.next,
+      size: 1
+    })
+    expect(resultPage2.data.length).toStrictEqual(0)
+  })
+
+  it('should listProposals with status, type and pagination', async () => {
+    const resultPage1 = await controller.listProposals(ListProposalsStatus.VOTING, ListProposalsType.CFP, undefined, {
+      size: 1
+    })
+    expect(resultPage1.data.length).toStrictEqual(1)
+    const resultPage2 = await controller.listProposals(ListProposalsStatus.VOTING, ListProposalsType.CFP, undefined, {
+      next: resultPage1.page?.next,
+      size: 1
+    })
+    expect(resultPage2.data.length).toStrictEqual(0)
+  })
+
+  // Get single related tests
+  it('should getProposal for CFP', async () => {
+    const result = await controller.getProposal(cfpProposalId)
+    expect(result).toStrictEqual({
+      proposalId: cfpProposalId,
+      creationHeight: expect.any(Number),
+      title: 'CFP proposal',
+      context: 'github',
+      contextHash: '',
+      status: GovernanceProposalStatus.VOTING,
+      type: GovernanceProposalType.COMMUNITY_FUND_PROPOSAL,
+      amount: new BigNumber(1.23).toFixed(8),
+      payoutAddress: payoutAddress,
+      currentCycle: 1,
+      totalCycles: 2,
+      cycleEndHeight: expect.any(Number),
+      proposalEndHeight: expect.any(Number),
+      votingPeriod: expect.any(Number),
+      quorum: expect.any(String),
+      approvalThreshold: expect.any(String),
+      fee: expect.any(Number)
+    })
+  })
+
+  it('should getProposal for VOC', async () => {
+    const result = await controller.getProposal(vocProposalId)
+    expect(result).toStrictEqual({
+      proposalId: vocProposalId,
+      creationHeight: 104,
+      title: 'VOC proposal',
+      context: 'github',
+      contextHash: '',
+      status: GovernanceProposalStatus.VOTING,
+      type: GovernanceProposalType.VOTE_OF_CONFIDENCE,
+      currentCycle: 1,
+      totalCycles: 1,
+      cycleEndHeight: expect.any(Number),
+      proposalEndHeight: expect.any(Number),
+      votingPeriod: expect.any(Number),
+      quorum: expect.any(String),
+      approvalThreshold: expect.any(String),
+      fee: expect.any(Number),
+      amount: undefined
+    })
+  })
+
+  // List vote related tests
+  it('should listProposalVotes', async () => {
+    const masternodeId = await getVotableMasternodeId()
+    await testing.rpc.governance.voteGov({
+      proposalId: cfpProposalId,
+      masternodeId: masternodeId,
+      decision: VoteDecision.NEUTRAL
+    })
+    await testing.container.generate(1)
+    const result = await controller.listProposalVotes(cfpProposalId)
+    expect(result.data).toStrictEqual([
+      {
+        proposalId: cfpProposalId,
+        masternodeId: masternodeId,
+        cycle: 1,
+        vote: ProposalVoteResultType.NEUTRAL
+      }
+    ])
+  })
+
+  it.only('should listProposalVotes with all records when limit is 0', async () => {
```

Comment:
Accidental commit? I believe this skips the other tests

Reply:
🙏 updated

Reply:
@eli-lim for some reason my recent push is not reflecting on this PR, possibly due to my access changes after i accepted the invite into jellyfish org, do you mind checking the new one https://github.com/JellyfishSDK/jellyfish/pull/1970 , thanks!